app/src/main/java/sopt/org/millie/core/designsystem/component/SearchTextField.kt

```diff
@@ -0,0 +1,109 @@
+package sopt.org.millie.core.designsystem.component
+
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import sopt.org.millie.R
+import sopt.org.millie.core.designsystem.theme.MillieTheme
+import sopt.org.millie.core.util.noRippleClickable
+
+@Composable
+fun SearchTextField(
+    value: String,
+    onValueChange: (String) -> Unit,
+    modifier: Modifier = Modifier,
+    onCancelClick: () -> Unit = { onValueChange("") },
```

Comment:
```suggestion
    onCancelClick: () -> Unit,
```
물론 지금 코드도 문제 없지만 onCancelClick 내부에서 value를 직접 초기화하는 것보다 상위에서 함께 처리할 수 있도록 이벤트만 외부로 넘기면 더 재사용성이 높아질 것 같은데 어떻게 생각하시나용 ?! 근데 지금처럼 작성하는것두 갠찮아요 !! 이건 진짜 지민이 의견대루 하면댐 !!

Reply:
아 초기화 로직을 아예 상위로 넘기자는 말씀이시죠?? 좋아요!!

Reply:
이거 근데 저 궁금한거 하나 있습니다..!!
그 저 취소 아이콘이 항상 나오는게 아닌데 이거 필수 파라미터로 남겨놓으면
취소 아이콘이 안 나타나는 상황에서 이 클릭 처리도 빈 람다로 놔둬야 하는거 아닌가용?? 
그러면 이렇게 기본값 두는 것도 좋은 것 같다고 생각하는데 이것 관련해서 의견이 궁금합니다!

`onCancelClick: () -> Unit = {}`
그래서 요렇게 하는건 어떠세용??

Reply:
디쌤들께 여쭤봣을때 취소 아이콘은 바코드 아이콘이 잇든 없든 텍스트를 치면 항상 나온다고 하셧어요 !! 그래서 `onCancelClick: () -> Unit,` 이런식으로 작성하고 상위에서 처리해도 될 것 같다구 말씀드린 거엿어용 !!  소희 언니가 말해준대루 빈 람다로 기본값 두는 것두 괜찮구요 !!  사용 편의성을 위해서는 기본값 = {}를 주는 것도 좋을 것같아요잉  왜냐하면 대부분의 경우 그냥 텍스트만 지우면 됨이니까요!! 지민이 편한대루 하면 될것 같아용 !!

Reply:
입력 내용이 있으면 취소 아이콘이 항상 존재하고 이 아이콘을 클릭했을 때 입력값이 삭제되는건 필수 상황이라고 생각해서 필수 파라미터로 남겨뒀습니당